bindings-generator/src/methods.rs

```diff
@@ -387,7 +396,126 @@ pub(crate) fn generate_methods(
                 }
             }
         };
+
         result.extend(output);
+
+        generated.insert(
+            method_name.to_string(),
+            Generated {
+                icall,
+                icall_ty,
+                maybe_unsafe,
+                maybe_unsafe_reason,
+            },
+        );
+    }
+
+    for property in &class.properties {
+        if property.index < 0 || property.name.contains('/') {
+            continue;
+        }
+
+        let property_index = property.index;
+        let ty = Ty::from_src(&property.type_);
+
+        if let Some(Generated {
+            icall,
+            icall_ty,
+            maybe_unsafe,
+            maybe_unsafe_reason,
+        }) = generated.get(&property.getter)
+        {
+            let rusty_name = rust_safe_name(&property.name);
+            let rust_ret_type = ty.to_rust();
+
+            let method_bind_fetch = {
+                let method_table = format_ident!("{}MethodTable", class.name);
+                let rust_method_name = format_ident!("{}", property.getter);
+
+                quote! {
+                    let method_bind: *mut sys::godot_method_bind = #method_table::get(get_api()).#rust_method_name;
+                }
+            };
+
+            let doc_comment = docs
+                .and_then(|docs| {
+                    docs.get_class_method_desc(
+                        class.name.as_str(),
+                        &format!("get_{}", property.name),
+                    )
+                })
+                .unwrap_or("");
+
+            let recover = ret_recover(&ty, *icall_ty);
+
+            let output = quote! {
+                #[doc = #doc_comment]
+                #[doc = #maybe_unsafe_reason]
+                #[inline]
+                pub #maybe_unsafe fn #rusty_name(&self) -> #rust_ret_type {
+                    unsafe {
+                        #method_bind_fetch
+
+                        let ret = crate::icalls::#icall(method_bind, self.this.sys().as_ptr(), #property_index);
+
+                        #recover
+                    }
+                }
+            };
+
+            result.extend(output);
```

Comment:
There is now quite a bit of code duplication:
* methods
* getters
* setters

Would it make sense to extract these parts into a function? It might need a few closure parameters to extract names, but I could imagine it still makes things easier to understand. If yes, this change could gladly be a separate commit.

Reply:
I've tried naive method extraction, but it didn't work very well due to the amount of arguments. Nested functions also didn't really help with readability. A larger scale refactoring is probably necessary if we want to meaningfully improve the clarity of this part, and I didn't want to do that without some prior communication.

If you think that's OK, I'll see what I can do about it.

Reply:
I see, thanks for investigating.

It's not urgent -- if you want to do a refactoring, I'd support that, but it's also OK if we leave it for now. In either case, I think this could be done as a separate PR, to unblock this already 🙂